rlclientlib/live_model_impl.cc

```diff
@@ -170,6 +170,10 @@ namespace reinforcement_learning {
       return error_code::not_supported;
     }
 
+    if(_protocol_version == 1 && _model->model_type() != model_management::model_type_t::SLATES) {
```

Comment:
is this a check that can be done by the logger facade on initialization so that we don't have to leak the protocol version to the live model impl?

Reply:
It's done there as well, but the error message won't be as good as the one we can get here.

Reply:
We get an invalid_argument with empty message.